yard_extensions/define_property_handler.rb

```diff
@@ -10,36 +10,36 @@ class DefinePropertyHandler < YARD::Handlers::Ruby::Base
     params = statement.parameters(false).dup
 
     # Add all attributes
-    validated_attribute_names(params).each do |name|
-      namespace.attributes[scope][name] ||= SymbolHash[:read => nil, :write => nil]
+    name = validated_attribute_names(params)
 
-      # Show their methods as well
-      {:read => name, :write => "#{name}="}.each do |type, meth|
-        if type == :read ? read : write
-          o = MethodObject.new(namespace, meth, scope)
-          if type == :write
-            o.parameters = [['value', nil]]
-            src = "def #{meth}(value)"
-            full_src = "#{src}\n  @#{name} = value\nend"
-            doc = "Sets the attribute #{name}\n@param value the value to set the attribute #{name} to."
-          else
-            src = "def #{meth}"
-            full_src = "#{src}\n  @#{name}\nend"
-            doc = "Returns the value of attribute #{name}"
-          end
-          o.source ||= full_src
-          o.signature ||= src
-          register(o)
-          o.docstring = doc if o.docstring.blank?(false)
+    namespace.attributes[scope][name] ||= SymbolHash[:read => nil, :write => nil]
 
-          # Register the object explicitly
-          namespace.attributes[scope][name][type] = o
+    # Show their methods as well
+    {:read => name, :write => "#{name}="}.each do |type, meth|
+      if type == :read ? read : write
```

Comment:
This line is confusing - too many conditionals in one go

Reply:
Yep, it is confusing :)

File overall is based on https://github.com/lsegal/yard/blob/master/lib/yard/handlers/ruby/attribute_handler.rb - I kept things the same where possible.

Reply:
Ah that makes more sense, I didn't think it was really your style. I leave it to your judgement what you want to do (if anything at all).